src/cryptsetup/cryptsetup-fido2.c

```diff
@@ -194,6 +198,17 @@ int find_fido2_auto_data(
 
                         pin_required = json_variant_boolean(w);
                 }
+
+                w = json_variant_by_key(v, "fido2-up-required");
+                if (w) {
+                        /* The "fido2-up-required" field is optional. */
+
+                        if (!json_variant_is_boolean(w))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "FIDO2 token data's 'fido2-up-required' field is not a boolean.");
+
```

Comment:
maybe ad "else" here

Reply:
sorry, not following - there's a return, and we usually don't use `else` to reduce indentation, no?

Reply:
ignore me. i am confused.